pkg/spdk/backing_image.go

```diff
@@ -0,0 +1,713 @@
+	}
+	return nil
+}
+
+func (bi *BackingImage) prepareFromSync(targetFh *os.File, fromAddress, srcLvsUUID string) (err error) {
+	if fromAddress == "" || srcLvsUUID == "" {
+		return errors.Wrapf(err, "missing required source backing image service address %v or source lvsUUID %v", fromAddress, srcLvsUUID)
+	}
+	srcBackingImageServiceCli, err := GetServiceClient(fromAddress)
+	if err != nil {
+		return errors.Wrapf(err, "failed to init the source backing image spdk service client")
+	}
+	exposedSnapshotLvolAddress, err := srcBackingImageServiceCli.BackingImageExpose(bi.Name, srcLvsUUID)
+	if err != nil {
+		return errors.Wrapf(err, "failed to expose the source backing image %v", bi.Name)
+	}
+	externalSnapshotLvolName := GetBackingImageSnapLvolName(bi.Name, srcLvsUUID)
+
+	srcIP, srcPort, err := splitHostPort(exposedSnapshotLvolAddress)
+	if err != nil {
+		return errors.Wrapf(err, "failed to split host and port from address %v", exposedSnapshotLvolAddress)
+	}
+	_, _, err = connectNVMeTarget(srcIP, srcPort, maxNumRetries, retryInterval)
+	if err != nil {
+		return errors.Wrapf(err, "failed to connect to NVMe target for source backing image %v in lvsUUID %v with address %v", bi.Name, srcLvsUUID, exposedSnapshotLvolAddress)
+	}
+
+	bi.log.Info("Creating NVMe initiator for source backing image %v", bi.Name)
+	initiator, err := nvme.NewInitiator(externalSnapshotLvolName, helpertypes.GetNQN(externalSnapshotLvolName), nvme.HostProc)
+	if err != nil {
+		return errors.Wrapf(err, "failed to create NVMe initiator for source backing image %v in lvsUUID %v with address %v", bi.Name, srcLvsUUID, exposedSnapshotLvolAddress)
+	}
+	if _, err := initiator.Start(srcIP, strconv.Itoa(int(srcPort)), true); err != nil {
+		return errors.Wrapf(err, "failed to start NVMe initiator for source backing image %v in lvsUUID %v with address %v", bi.Name, srcLvsUUID, exposedSnapshotLvolAddress)
+	}
+
+	bi.log.Infof("Opening NVMe device %v", initiator.Endpoint)
+	srcFh, err := os.OpenFile(initiator.Endpoint, os.O_RDWR, 0666)
+	if err != nil {
+		return errors.Wrapf(err, "failed to open NVMe device %v for source backing image %v in lvsUUID %v with address %v", initiator.Endpoint, bi.Name, srcLvsUUID, exposedSnapshotLvolAddress)
+	}
+
+	ctx, cancel := context.WithCancel(bi.ctx)
+	defer cancel()
+	_, err = util.IdleTimeoutCopy(ctx, cancel, srcFh, targetFh, bi, true)
+	if err != nil {
+		return errors.Wrapf(err, "failed to copy the source backing image %v in lvsUUID %v with address %v", bi.Name, srcLvsUUID, exposedSnapshotLvolAddress)
+	}
+
+	// Stop the source initiator
+	srcFh.Close()
+	bi.log.Info("Stopping NVMe initiator")
+	if _, err := initiator.Stop(true, true, false); err != nil {
+		return errors.Wrapf(err, "failed to stop NVMe initiator")
+	}
+
+	if err := srcBackingImageServiceCli.BackingImageUnexpose(bi.Name, srcLvsUUID); err != nil {
+		return errors.Wrapf(err, "failed to unsexpose the source backing image %v", bi.Name)
+	}
+
```

Comment:
I'm think if we can move the close, stop and unexpose functions to a derfer function. WDYT?

Reply:
Moved, it should